vault/logical_system.go

```diff
@@ -635,7 +660,64 @@ type SystemBackend struct {
 	Backend *framework.Backend
 }
 
-// handleCapabilitiesreturns the ACL capabilities of the token for a given path
+// corsStatus returns the current CORS configuration as a logical.Response
+func (b *SystemBackend) corsStatusResponse() (*logical.Response, error) {
+	corsConf := b.Core.corsConfig
+	if corsConf == nil {
+		return nil, errCORSNotConfigured
+	}
+
+	return &logical.Response{
+		Data: map[string]interface{}{
+			"enabled":         corsConf.Enabled,
+			"allowed_origins": strings.Join(corsConf.AllowedOrigins, " "),
+		},
+	}, nil
+}
+
+// handleCORSEnable sets the list of origins that are allowed
+// to make cross-origin requests and sets the CORS enabled flag to true
+func (b *SystemBackend) handleCORSEnable(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	origins := d.Get("allowed_origins").(string)
+
+	err := b.Core.corsConfig.Enable(origins)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &Config{
+		Name: "cors",
+		Settings: map[string]string{
+			"allowed_origins": origins,
+			"enabled":         "true",
+		},
+	}
+
+	// Update the config
+	if err := b.Core.configStore.SetConfig(config); err != nil {
+		return handleError(err)
+	}
+
+	return b.corsStatusResponse()
+}
+
+// handleCORSDisable clears the allowed origins and sets the CORS enabled flag to false
+func (b *SystemBackend) handleCORSDisable(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	b.Core.CORSConfig().Disable()
+
+	if err := b.Core.configStore.DeleteConfig("cors"); err != nil {
+		return handleError(err)
+	}
+
+	return b.corsStatusResponse()
```

Comment:
There is no need to return the values during deletion. This could just be `return nil, nil`.